DexcomCGMKitUI/ViewControllers/DexcomUICoordinator.swift

```diff
@@ -39,19 +39,30 @@ enum DexcomUIScreen {
     case transmitterIDEntryPhoto
     case insertSensorVideo
     
-    func next() -> DexcomUIScreen? {
-        // for the setup workflow, this assumes both the sensor and transmitter are being replaced. If otherwise, the ReplacementWorkflowViewModel needs to handle it
+    func next(_ workflow: ReplacementWorkflow) -> DexcomUIScreen? {
         switch self {
         case .sensorStop:
             return .sensorCodeLocation
         case .sensorCodeLocation:
             return .sensorCodeEntryOptions
         case .sensorCodeNone:
-            return .transmitterIDEntryOptions
+            if workflow == .replaceTransmitterAndSensor {
+                return .transmitterIDEntryOptions
+            } else {
+                return .sensorInsert
+            }
```

Comment:
[nit] consider a `switch` statement here (and below) with explicit `case`s:
```suggestion
            switch workflow  {
            case .replaceTransmitterAndSensor: return .transmitterIDEntryOptions
            case .replaceSensor: return .sensorInsert
            }
```